adagrams/game.py

```diff
@@ -1,11 +1,212 @@
+#       -Imagine that the user returns their hand to the pool before 
+#       drawing new letters
+
 def draw_letters():
-    pass
+    # Create a dictionary with key:value paris as letter:# of letters available
+    LETTER_POOL = {
+        'A': 9, 
+        'B': 2, 
+        'C': 2, 
+        'D': 4, 
+        'E': 12, 
+        'F': 2, 
+        'G': 3, 
+        'H': 2, 
+        'I': 9, 
+        'J': 1, 
+        'K': 1, 
+        'L': 4, 
+        'M': 2, 
+        'N': 6, 
+        'O': 8, 
+        'P': 2, 
+        'Q': 1, 
+        'R': 6, 
+        'S': 4, 
+        'T': 6, 
+        'U': 4, 
+        'V': 2, 
+        'W': 2, 
+        'X': 1, 
+        'Y': 2, 
+        'Z': 1
+    }
+    # Empty list of drawn_letters
+    drawn_letters = []
+    available_letters = []
+    # List key*value for each key:value pairs ---> returns the number 
+    #       of letters in a list that is available. Each letter is an element in the list
+    for letter in LETTER_POOL.items():
+        for count in range(letter[1]):
+            available_letters.append(letter[0])
+    # Select a random integer (0-len(list)) to select the letter
+    # .pop the list when letter is pulled (based on index value) so that only 
+    #       remaining letters are available
+    for i in range(10):
+        random_index = random.randint(0, len(available_letters)-1)
+        letter_pulled = available_letters.pop(random_index)
+        drawn_letters.append(letter_pulled)
+    # return drawn_letters
+    return drawn_letters
+
+
+
+
+#-------------------------------------------------------------------
+#WAVE 02
+# [ ] Has two parameters:
+#       -word, the first parameter, describes some input word, and is 
+#       a string
+#       -letter_bank, the second parameter, describes an array of drawn 
+#       letters in a hand. You can expect this to be an array of ten 
+#       strings, with each string representing a letter
+# [ ] Returns either true or false
+# [ ] Returns true if every letter in the input word is available 
+#     (in the right quantities) in the letter_bank
+# [ ] Returns false if not; if there is a letter in input that is not present 
+#     in the letter_bank or has too much of compared to the letter_bank
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # Create list that is a copy of letter_bank? Loop through to make the list?
+    letter_bank_copy = []
+    for letter in letter_bank:
+        letter_bank_copy.append(letter)
+
+    # for loop to check letters in word in word_bank
+    valid_letter = True
+    while valid_letter:
+        for letter in word.upper():
+        # if statement to determine TRUE/FALSE for validity (letter in bank, not if real word)
+            if letter in letter_bank_copy:
+                letter_bank_copy.remove(letter)
+                valid_letter = True
+            else:
+                valid_letter = False
+                return False
+        return True
```

Comment:
Not that the while loop is not necessary here. You return out of the entire function (you don't just break out of the loop) when you find a letter that's not valid. Consider this refactor: 
```suggestion
    for letter in word.upper():
    # if statement to determine TRUE/FALSE for validity (letter in bank, not if real word)
            if letter in letter_bank_copy:
                letter_bank_copy.remove(letter)
            else:
                return False
    return True
```